src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,81 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.UserRegisterException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceTest {
+
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_ok() {
+        User newUser = new User();
+        newUser.setLogin("anonym");
+        newUser.setPassword("123456");
+        newUser.setAge(18);
+        User userFromDb = registrationService.register(newUser);
+
+        assertNotNull(userFromDb.getId());
+        assertEquals(newUser.getLogin(), userFromDb.getLogin());
+    }
+
+    @Test
+    void register_exist_notOk() {
+        User newUser = new User();
+        newUser.setLogin("abcdef");
+        newUser.setPassword("123456");
+        newUser.setAge(19);
+        registrationService.register(newUser); // добавляем пользователя
+        User sameLoginUser = new User();
+        sameLoginUser.setLogin("abcdef"); // тот же логин
+        sameLoginUser.setPassword("654321");
+        sameLoginUser.setAge(20);
+
+        assertThrows(UserRegisterException.class,
+                () -> registrationService.register(sameLoginUser));
+    }
```

Comment:
The test `register_exist_notOk` is not complete. You need to add the first user to the storage directly using `Storage.people.add()` instead of `RegistrationService.register()` method. Because we can not use the same method which is under a test to insert some mock test data into our storage since we can not guarantee its correctness yet.

Reply:
It isn't part of this test!